### PR TITLE
Add log-format option to CLI

### LINKED ISSUE #740: Add option to log as JSON.
## Description

Based on https://github.com/sigp/lighthouse/pull/414, it is possible to log to a file, in which case logs are formatted with JSON, but it should be possible to do so for std out as well, as many log ingestion systems get logs from there and would want them structured.

## Present Behaviour

Only possible to output json logs to a file.

## Expected Behaviour

Option to format logs as json regardless of destination.

### PR DESCRIPTION
**Pull request by pscott**
## Issue Addressed

Closes #740 

## Proposed Changes

- Add `--log-format` option to enable users to specify their desired logging format.
- Add appropriate documentation

## Additional Info

This option currently only supports the `JSON` format but I've set up the code to easily be able to add other formats.

### CODE REVIEW

File: lighthouse/environment/src/lib.rs [hunk 1]
```diff
@@ -256,8 +276,14 @@ impl<E: EthSpec> Environment<E> {
             .open(&path)
             .map_err(|e| format!("Unable to open logfile: {:?}", e))?;
 
-        let drain = Mutex::new(slog_json::Json::default(file)).fuse();
-        let drain = slog_async::Async::new(drain).build();
+        let log_format = log_format.unwrap_or("JSON");
+        let drain = match log_format.to_uppercase().as_str() {
+            "JSON" => {
+                let drain = Mutex::new(slog_json::Json::default(file)).fuse();
+                slog_async::Async::new(drain).build()
```
Comment (pscott) [inline lighthouse/environment/src/lib.rs]: The initial logger here had this code. Wondering if Mutex is really necessary here? I think we could just do 
```rust
let drain = slog_json::Json::default(file)).fuse;
slog_async::Async::new(drain).build()
```

Comment (paulhauner) [reply lighthouse/environment/src/lib.rs]: Hmm.. it's not clear to me why this is a `Mutex`. Does it compile without the Mutex?

IIRC @kirk-baird added the Mutex originally.

Comment (pscott) [reply lighthouse/environment/src/lib.rs]: Yup, it compiles without it. I asked just to know if there was a specific reason to the Mutex :) I added a new commit that removes the mutex!

Comment (kirk-baird) [reply lighthouse/environment/src/lib.rs]: The tutorial I was following used a mutex, having a look through it now the mutex is unnecessary so happy for it to be removed :)

**Pull request merged**